src/include/expression/expression_util.h

```diff
@@ -101,6 +101,87 @@ class ExpressionUtil {
     }
   }
 
+  /*
+   * This function removes all the terms related to indexed columns within an
+   * expression.
+   *
+   * NOTE: This should only be called after we check that the predicate is index
+   * searchable. That means there are only "and" conjunctions.
+   */
+  static AbstractExpression *RemoveTermsWithIndexedColumns(
+      AbstractExpression *expression, std::shared_ptr<index::Index> index) {
+    LOG_TRACE("Expression Type --> %s",
+              ExpressionTypeToString(expression->GetExpressionType()).c_str());
+
+    size_t children_size = expression->GetChildrenSize();
+    // We shouldn't have a expression that has more than two children
+    PL_ASSERT(children_size <= 2);
```

Comment:
Ask @MattPerron about this. I think that you can have more than two children for some functions.